### PR TITLE
Added tests for getting game lists using serializers and the api-game…

### LINKED ISSUE #327: Creating Test Cases: Game GET API Endpoint
This issue pertains to the development of comprehensive test cases for the Game GET API Endpoint. The objective is to ensure that the endpoint reliably returns accurate game data under various scenarios.

### PR DESCRIPTION
**Pull request by giomhern**
This is the pull request associated with issue #327 on creating more advanced and adequate checks are made on the Game List and that the Games produced with the factories are the same as those from the response.

### CODE REVIEW

**Review by cuyakonwu — APPROVED** [review]
Looks great, and is passing well!

**Review by cuyakonwu — APPROVED** [review]
Passes all tests and looks good!

File: src/chigame/api/tests/test_api.py [hunk 1]
```diff
@@ -17,7 +21,14 @@ def check_equal(self, obj, expected: dict):
         Helper function to check that the object data matches the expected data.
         """
         for key in expected:
-            self.assertEqual(getattr(obj, key), expected[key])
+            if isinstance(obj, (ReturnDict, dict)):
+                # Use key indexing for dictionaries and ReturnDict
```
Comment (majorsylvie) [inline src/chigame/api/tests/test_api.py]: This should include a why in the code comment. 
Why use/need key indexing for what attriutes.

**Review by majorsylvie — CHANGES_REQUESTED** [review]
Nice additions, appreciate the import comments, think they work well.

Please add more body to a code comment but otherwise this is good to merge!

File: src/chigame/api/tests/test_api.py [hunk 2]
```diff
@@ -17,7 +21,24 @@ def check_equal(self, obj, expected: dict):
...
+            # Issue:  Serialized data often converts numbers to strings for transport
+            # leading to a type mismatch when compared with their original Python types.
+
+            # Solution: The check on lines 35-36 converts the serialized data to
+            # the original Python type before comparing it with the expected data.
+            # This ensures that the comparison is done on the same type of data.
+
+            # For example, without this check, you might encounter issues
+            # when comparing Decimal('5') and '5.00',
+            # which would fail due to type mismatch despite representing the same value.
```
Comment (majorsylvie) [inline src/chigame/api/tests/test_api.py:33]: Absolutely fantastic code comment.
Clear, direct, justified, with an example to boot!

Thank you for your work

**Review by majorsylvie — APPROVED** [review]
Fantastic code comment! 

Thank you for adding more tests.

With feedback addressed this is good to go.

Thank you for your work

**Comment by giomhern** [comment]
This PR, rooted in Issue #327, was focused on enhancing and ensuring that our GET API endpoint for games and the associated factory worked up to standards. Previously, we had an issue of mismatched values within the serialized and unserialized data which would cause the tests to fail. This has now been resolved in the helper function ``` check_equal ``` and the tests pass wonderfully!

Thank you to @giomhern for coding up the tests and @cuyakonwu for the help reviewing!

**Comment by majorsylvie** [comment]
Issue Score: Excellent

Comments:
Great work!
Very very good closing comment.

**Pull request merged**